modules/ggipc-client/src/client.c

```diff
@@ -239,23 +279,13 @@ GglError ggipc_call(
         return GGL_ERR_FAILURE;
     }
 
-    if (common_headers.message_type == EVENTSTREAM_APPLICATION_ERROR) {
-        GGL_LOGI(
-            "Received IPC error on stream %" PRId32 ". Payload: %.*s.",
-            common_headers.stream_id,
-            (int) msg.payload.len,
-            msg.payload.data
-        );
-        return GGL_ERR_FAILURE;
-    }
-
-    if (common_headers.message_type != EVENTSTREAM_APPLICATION_MESSAGE) {
-        GGL_LOGE(
-            "Stream %" PRId32 " received invalid message type: %" PRId32 ".",
-            common_headers.stream_id,
-            common_headers.message_type
-        );
-        return GGL_ERR_FAILURE;
+    if (common_headers.content_type.data != NULL) {
```

Comment:
We always expect a payload for application message responses, so this should be required.